### PR TITLE
ISPN-11257 Indexed queries should automatically be broadcasted

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11257

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/SearchManager.java [hunk 1]
```diff
@@ -21,6 +21,13 @@
     */
    <E> CacheQuery<E> getQuery(String queryString, IndexedQueryMode indexedQueryMode);
 
+   /**
+    * Builds a {@link CacheQuery} from an Ickle query string, assuming the correct value for {@link IndexedQueryMode} to
+    * query all data in the cluster
+    * @param queryString the Ickle query
+    * @return
+    */
+   <E> CacheQuery<E> getQuery(String queryString);
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/SearchManager.java]: no blank line? blasphemy

Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/SearchManager.java]: Fixed

File: query/src/main/java/org/infinispan/query/impl/IndexInspector.java [hunk 2]
```diff
@@ -61,6 +61,10 @@ public IndexInspector(Configuration cfg, SearchIntegrator searchFactory) {
       return indexManagerPerClass;
    }
 
+   public IndexManager get(Class<?> c) {
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/impl/IndexInspector.java]: This one does not seem the be used anywhere. The name is also a bit non-descriptive. Maybe we could call it getIndexManager, or getIndexManagerForClass, or remove it if really not used.

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java [hunk 3]
```diff
@@ -544,57 +542,57 @@ public void testClear() throws Exception {
...
-      assertEquals("ExtraGoat", person.getName());
-      assertEquals(70, person.getAge());
-
-      //Disabling the fullTextFilter.
-      query.disableFullTextFilter("personFilter");
-      query.disableFullTextFilter("personAgeFilter");
-      assertEquals(3, query.getResultSize());
-      StaticTestingErrorHandler.assertAllGood(cache1, cache2);
-   }
+//   public void testFullTextFilterOnOff() throws Exception {
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java]: Are these tests obliterated for now with the hope to resurrect them later ?

Comment (gustavonalle) [reply query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java]: This is something that leaks from Hibernate Search API: the capacity to annotate a filter in the entity and enable or disable it at query time. In the Ickle world, the equivalent would be to not use a predicate when not required. WDYT?

File: query/src/main/java/org/infinispan/query/impl/IndexInspector.java (hunk 2 above)
Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/impl/IndexInspector.java]: I will double check

File: server/rest/src/test/java/org/infinispan/rest/search/NonSharedIndexSearchTest.java [hunk 4]
```diff
@@ -22,8 +22,8 @@ ConfigurationBuilder getConfigBuilder() {
       return builder;
    }
 
-   @Override
```
Comment (anistor) [inline server/rest/src/test/java/org/infinispan/rest/search/NonSharedIndexSearchTest.java]: It seems that removing ```@Override``` was not really needed.

Comment (gustavonalle) [reply server/rest/src/test/java/org/infinispan/rest/search/NonSharedIndexSearchTest.java]: fixed

File: query/src/main/java/org/infinispan/query/impl/IndexInspector.java (hunk 2 above)
Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/impl/IndexInspector.java]: This was not used. It's probably a leftover

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java (hunk 3 above)
Comment (anistor) [reply query/src/test/java/org/infinispan/query/blackbox/ClusteredCacheTest.java]: Indeed. let's leave them commented out for now. I have no idea what to do with them.

**Comment by anistor** [comment]
I've fixed some more small ws issues and merged. Thanks @gustavonalle !

**Pull request closed without merging**